src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -1128,6 +1171,153 @@ class CreateStatementSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment()
+class CreateExternalTableSegment(BaseSegment):
+    """A snowflake `CREATE EXTERNAL TABLE` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-external-table.html
+    """
+
+    type = "create_external_table_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        Sequence("OR", "REPLACE", optional=True),
+        "EXTERNAL",
+        "TABLE",
+        Sequence("IF", "NOT", "EXISTS", optional=True),
+        Ref("TableReferenceSegment"),
+        # Columns:
+        Sequence(
+            Bracketed(
+                Delimited(
+                    OneOf(
+                        Ref("TableConstraintSegment"),
+                        Ref("ColumnDefinitionSegment"),
+                    ),
+                ),
+            ),
+            optional=True,
+        ),
+        AnyNumberOf(
+            Sequence(
+                "PARTITION",
+                "BY",
+                Delimited(
+                    Ref("SingleIdentifierGrammar"),
+                ),
+                optional=True,
+            ),
+            Sequence(
+                Sequence("WITH", optional=True),
+                "LOCATION",
+                Ref("EqualsSegment"),
+                Ref("AtSignLiteralSegment"),
+                Bracketed(
+                    Ref("NakedIdentifierSegment"),
```

Comment:
I'm not familiar with Snowflake, but this seems odd -- just an identifier and a dot in square brackets?

Reply:
It's because it's an optional namespace, so this is correct - it just needs the identifier following this!

```
 externalStage ::=
  @[<namespace>.]<ext_stage_name>[/<path>]
```